src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +501,128 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isVerbose)
+        {
+            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, "out-of-process");
+                }
+                return false;
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (await ShouldNet8InProcBeLaunched())
+                {
+                    if (isVerbose)
+                    {
+                        PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
+                    }
+                    return true;
+                }
+                else
+                {
+                    throw new CliException($"Invalid config for running {DotnetConstants.InProc8HostRuntime} host.");
+                }
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                }
+                return true;
+            }
+            else
+            {
+                throw new CliException($"Invalid host runtime '{HostRuntime}'. Valid values are 'default', '{DotnetConstants.InProc8HostRuntime}', '{DotnetConstants.InProc6HostRuntime}'.");
+            }
+        }
+
+        private void PrintVerboseForHostSelection(bool isVerbose, string hostRuntime)
+        {
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Selected {hostRuntime} host"));
+            }
+        }
+
+        private async Task<bool> ShouldNet8InProcBeLaunched()
+        {
+            // Start .NET 8 child process if InProc8 is enabled
+            if (ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
```

Comment:
Do we need this `ShouldLaunchInProcNet8AsChildProcess` method anymore? It was needed when we had a single branch, and we were producing artifacts for 2 versions (net6,net8). For this branch (oop core tools), I do not think we need that method anymore.

Reply:
removed method